src/datachain/catalog/catalog.py

```diff
@@ -1198,40 +1252,77 @@ def get_remote_dataset(
 
     def get_dataset_dependencies(
         self,
-        name: str,
-        version: str,
+        name: str | None = None,
+        version: str | None = None,
         namespace_name: str | None = None,
         project_name: str | None = None,
         indirect=False,
+        dataset_id: int | None = None,
+        dataset_version_id: int | None = None,
     ) -> list[DatasetDependency | None]:
```

Comment:
Not directly related to this PR, but I am still not sure if this is a good return type: `list[DatasetDependency | None]` 😢 Does it make any sense to get Nones in the list of dependencies.